### ISSUE TITLE
Add all words in the current document to dictionary

### ISSUE THREAD

**Issue by laggingreflex** [Jason-Rev/vscode-spell-checker#59]
Please add an option to add all words in the current document (that are deemed incorrect) to the project/global/custom dictionary.

**Comment by Jason3S** [comment]
I understand the desire for this feature.  At the same time, I would like to get better idea of what issue you are trying to address.  Are you seeing a LOT of words that do not match?  Does another language needed to be added?  Or is it necessary to turn on compound words?

**Comment by laggingreflex** [comment]
I use a lot of other people's projects that have words that are flagged as incorrect and I don't necessarily need to, or even can, correct them, so I would rather wish to add all those words in the project dictionary. Some projects also contain words that I would like to add to my user dictionary as well. 

Often times there are a LOT of words, I currently have to do this for each word. 

My main intention is to clear all spelling related annotations over the vertical scroll bar on the right so it doesn't interfere with other features of it, like code changes, breakpoints, errors etc.

**Comment by Jason3S** [comment]
Just curious, which programming language do you use?

The easiest way to add this feature is to make it an F1 command.  Like: `F1` + `add all word to dictionary`.  Which will show a possible set of commands to pick from.

**Comment by laggingreflex** [comment]
Sounds good. I mainly use Javascript.

**Comment by hermanbergwerf** [comment]
This would be very useful for me too. A lot of my variable names are flagged (I use abbreviations, such as `ivec` for the i unit vector). I usually disable spellcheck again after a while because it distracts me (flags too much words, mixes with linting errors etc.), but this feature would make it possible to quickly create your own 'dictionary'. (just installed it, already got a couple of spelling mistakes in my comments!).

**Comment by lucasbento** [comment]
Just an addendum to this feature, would be nice to have a right-click-menu item named `Add this word to dictionary` that shows up when you select a word.

**Comment by Jason3S** [comment]
@lucasbento good idea.

**Comment by Jason3S** [comment]
I have taken the suggestions and added this feature in the following way:
1. Select the text you want to add to the word list.
2. Right click on the text and choose which word list to add it to:
    - A Word to Dictionary
    - A Word to Global Dictionary

The extension will split the words and add them to the chosen dictionary.

**Comment by shenlebantongying** [comment]
Thank god!

This design is so good!

**Comment by laggingreflex** [comment]
@Jason3S Sorry for coming back to this after such a long time. Thanks for implementing it. I just tested it in a couple new projects. It's proving very useful indeed. 

But I have a concern. 

This  seems to add **all words** in the selected text to the dictionary. While it does solve the problem, but it seems a bit excessive, no? A typical code file may have thousands of words. Most of them already recognized by the default dictionary, so adding them is redundant. It may also have performance impact as the dictionary grows. 

I currently, in all my usage of this plugin over the years, have accumulated ~600 custom words. But using this to add all words in a single file of a new project almost doubled it. Although the list does plateaus after adding the most used words in English. 

It's fine for now, and does the job well, but I think it'd be nice to improve this feature if possible by only adding the custom words.

**Comment by mortezaataiy** [comment]
At lest please support that in PROBLEMS section. When select multiple word we cant add all of them to dictionary.
![image](https://user-images.githubusercontent.com/17705366/149652046-55295b95-48bd-4ee7-a322-fab0866ce4fc.png)

**Comment by mortezaataiy** [comment]
Or some time we are starting an existing big project and old words is not important. This feature is very useful in this situation. Thanks ❤

**Comment by Jason3S** [comment]
@mortezaataiy,

Something to try. Use the [cspell](https://www.npmjs.com/package/cspell) command line tool to get the list of words for a new project.

The following will search all files (the `"**"`) and put the one it found into a file called `words-found.txt`.

```sh
npx cspell "**" --no-problems --no-progress --unique > words-found.txt
```

The code spell checker extension uses the CSpell Library to do the spell checking. Which means both `cspell` and this extension will use any settings found in `cspell.json` files, hopefully making things easier.
